daml-lf/kv-support/src/test/scala/com/daml/lf/kv/transactions/TransactionTraversalSpec.scala

```diff
@@ -88,14 +47,48 @@ class TransactionTraversalSpec extends AnyFunSuite with Matchers {
   }
 
   test("extractPerPackageWitnesses - extract package witness mapping as expected") {
-    val result = TransactionTraversal.extractPerPackageWitnesses(rawTx)
+    val testTransaction = RawTransactionForTest()
+    val result = TransactionTraversal.extractPerPackageWitnesses(testTransaction.rawTx)
+    result shouldBe
+      Right(
+        Map(
+          "template_exercise" -> Set("Alice", "Charlie"),
+          "interface_exercise" -> Set("Alice", "Charlie"),
+          "template_create" -> Set("Alice", "Charlie", "Bob"),
+          "template_fetch" -> Set("Alice", "Bob"),
+        )
+      )
+  }
+
+  test(
+    "extractPerPackageWitnesses - extract package witness mapping as expected including rollback node"
+  ) {
+    val testTransaction = RawTransactionForTest()
+    val lookupByKeyNode = withNodeBuilder { builder =>
+      builder.setLookupByKey(
+        NodeLookupByKey.newBuilder
+          .setTemplateId(identifierForTemplateId("template_lookup_by_key"))
+          .setKeyWithMaintainers(
+            KeyWithMaintainers.newBuilder().addMaintainers("New maintainer")
+          )
+      )
+    }
+    testTransaction.builder.addRoot(withNodeBuilder { builder =>
+      builder.setRollback(
```

Comment:
you could consider testing nested rollbacks node although I think in this particular case it doesn’t add much.